addons/website_sale/models/product.py

```diff
@@ -85,118 +58,118 @@ def _inverse_image_small(self):
+    website_sequence = fields.Integer(string='Sequence', help="Determine the display order in the Website E-commerce",
+                                      default=lambda self: self._default_website_sequence())
+    public_categ_ids = fields.Many2many('product.public.category', string='Website Product Category', help="Those categories are used to group similar products for e-commerce.")
+
+    def _default_website_sequence(self):
+        min_sequence = self.search([], order='website_sequence', limit=1).website_sequence
+        return min_sequence and min_sequence - 1 or 10
+
+    def set_sequence_top(self):
+        self.website_sequence = self.search([], order='website_sequence desc', limit=1).website_sequence + 1
+
+    def set_sequence_bottom(self):
+        self.website_sequence = self.search([], order='website_sequence', limit=1).website_sequence - 1
+
+    def set_sequence_up(self):
+        previous_product_tmpl = self.search([('website_sequence', '>', self.website_sequence), ('website_published', '=', self.website_published)], order='website_sequence', limit=1)
+        if previous_product_tmpl:
+            previous_product_tmpl.website_sequence, self.website_sequence = self.website_sequence, previous_product_tmpl.website_sequence
         else:
-            return self.set_sequence_top(cr, uid, ids, context=context)
-
-    def set_sequence_down(self, cr, uid, ids, context=None):
-        product = self.browse(cr, uid, ids[0], context=context)
-        cr.execute("""  SELECT id, website_sequence FROM product_template
-                        WHERE website_sequence < %s AND website_published = %s ORDER BY website_sequence DESC LIMIT 1""" % (product.website_sequence, product.website_published))
-        next = cr.fetchone()
-        if next:
-            self.write(cr, uid, [next[0]], {'website_sequence': product.website_sequence}, context=context)
-            return self.write(cr, uid, [ids[0]], {'website_sequence': next[1]}, context=context)
+            self.set_sequence_top()
+
+    def set_sequence_down(self):
+        next_prodcut_tmpl = self.search([('website_sequence', '<', self.website_sequence), ('website_published', '=', self.website_published)], order='website_sequence desc', limit=1)
+        if next_prodcut_tmpl:
+            next_prodcut_tmpl.website_sequence, self.website_sequence = self.website_sequence, next_prodcut_tmpl.website_sequence
         else:
-            return self.set_sequence_bottom(cr, uid, ids, context=context)
+            return self.set_sequence_bottom()
+
+    @api.multi
+    def _website_url(self, field_name, arg):
+        res = super(ProductTemplate, self)._website_url(field_name, arg)
+        for product in self:
+            res[product.id] = "/shop/product/%s" % (product.id,)
+        return res
 
 
-class product_product(osv.Model):
+class Product(models.Model):
     _inherit = "product.product"
 
+    # TO-REMOVE: we use website_publish_button instead of open_website_url
```

Comment:
@rga-odoo it's ok to remove code if:

1. you do it in a different commit (ok)
2. you say why in the **commit message**

You do not need to comment the code you want to remove (and you have left the original code `open_website_url` so it was duplicated code), just remove it. It's the point of using git, no need to comment code or explain why in comments (that's what commit messages are for)

(I will push the fix, no need to do it)